### ISSUE TITLE
Selecting and copying newline character

### ISSUE THREAD

**Issue by NickeZ** [jwilm/alacritty#432]
Hey all,

Selection still doesn't really behave like other terminals.

1. If I click on the first cell of a row and drag down I expect to also select and put the newline in the clipboard. It looks like it is doing that, the whole line is selected, but when I paste there isn't any newline. (btw, I don't want to select the first character of the second row.)
2. Selecting beyond the last character should also select all the way to the right side and put a newline character in the clipboard. Instead I can select empty cells.

**Comment by chrisduerr** [comment]
So recently I've played around a bit with the selection and noticed that when a selection is made and the mouse is dragged outside the right side of the alacritty window, the last column is not selected anymore.

This is due to the way the side calculation works in input.rs. The last column is only included when the cursor position goes over the half way point of the glyph. However when the cursor leaves the screen, the cursor position relative to the cell is reset to 0 and the side is Side::Left, even tho the mouse leaves the window at the right side.

I feel like this might also be the reason for #777, so it seems like the results of the issue were fixed, instead of the actual issue itself?
